### DISCUSSION TITLE
Chrome Extension store reviewers rejecting extensions with react-player

### DISCUSSION THREAD

**Discussion by AntonOfTheWoods** [cookpete/react-player#1787] [General]
#### Current Behavior
Store reviewers are rejecting extensions with `react-player` as a dependency

#### Expected Behavior
Store reviewers not to reject `react-player` as a dependency

#### Steps to Reproduce
1. Add as a dependency
1. Submit to extension store
1. Get angry at the rejection

#### Environment
- URL attempting to play: 
- Browser: Chrome
- Operating system: all

#### Other Information
In exchanges with the reviewer it appears that the "remote code" that causes the rejection are the links to the SDKs. That code is not executed - they just saw code that *could* be executed, and rejected. As my use of `react-player` is with a file and a custom player, I was able to get around the issue largely by copying the relevant code and removing all the SDK-based stuff.

It may also simply be an overzealous reviewer rather than a change in policy - I have had several submissions succeed so I don't know whether this is the new normal or not.
 
I understand this might be a direct wontfix but it might be useful to have this here if others get a rejection and can't work out why!

**Comment by animber0** [comment]
hi!

I have a general question about how you were even able to get this to load into an extension using ManifestV3 this gets an immediate CSP rejection even in the browser!

**Comment by AntonOfTheWoods** [comment]
@animber0 https://github.com/transcrobes/transcrobes/blob/main/frontend/manifest.config.ts worked and was accepted at previous points. I ended up internalising the code because I only needed a fraction of it, the project is basically finished and the store was rejecting.

**Comment by animber0** [comment]
> @animber0 https://github.com/transcrobes/transcrobes/blob/main/frontend/manifest.config.ts worked and was accepted at previous points. I ended up internalising the code because I only needed a fraction of it, the project is basically finished and the store was rejecting.

Do you have any insight into how one would be able to load the remote scripts? Like you pointed out they're pretty much the only things stopping this from being used as required. I tried the answer [here](https://github.com/CookPete/react-player/issues/853#issuecomment-616125788), but it only worked for vimeo and two other sources (and not for Youtube for example).

**Comment by AntonOfTheWoods** [comment]
@animber0 no idea, sorry, and things seem pretty slow here, so you might want to investigate forking for your own requirements.

**Comment by animber0** [comment]
> @animber0 no idea, sorry, and things seem pretty slow here, so you might want to investigate forking for your own requirements.

Forking it is!

**Comment by RobinLbt** [comment]
Hey @animber0 , I'm running into the same issue

Did you managed to fix it?
